architecture/src/commonMain/kotlin/observable/Observable.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.architecture.observable
+
+import com.splendo.kaluga.flow.BaseFlowable
+import kotlin.properties.ObservableProperty
+import kotlin.properties.ReadOnlyProperty
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Result type for an [Observable]. Used to allow for the distinction between `null` and optional values
+ */
+sealed class ObservableResult<T> : ReadOnlyProperty<Any?, T?> {
```

Comment:
Usually `Result` holding a returning value or an error, in case you have value or null, it's Maybe (Option) type

Reply:
Renamed to ObservableOptional